lib/MonadEq_Lemmas.thy

```diff
@@ -0,0 +1,278 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+(* Lemmas that support the monad_eq method. Either things that go into the [monad_eq] set
```

Comment:
potential idea for factoring a method out to its own file: say what it does?

Reply:
The method already has its own file that describes what the method does. This new file just has the supporting lemmas for the NonDetMonad for this method. I think this is right as written.

Reply:
Have added a comment that points to the MonadEq theory